### PR TITLE
Fix warnings

### PR DESCRIPTION
**Pull request by jcfr**

### CODE REVIEW

**Comment by jcfr** [comment]
Remaining warnings (reported using `g++ (Ubuntu 5.2.1-22ubuntu2) 5.2.1 20151010`) would have to be fixed in upstream DCMTK:

```
In file included from /home/jcfr/Projects/Slicer-Release/DCMTK/dcmiod/include/dcmtk/dcmiod/modcommoninstanceref.h:32:0,
                 from /home/jcfr/Projects/Slicer-Release/DCMTK/dcmiod/include/dcmtk/dcmiod/iodcommn.h:34,
                 from /home/jcfr/Projects/Slicer-Release/DCMTK/dcmiod/include/dcmtk/dcmiod/iodimage.h:28,
                 from /home/jcfr/Projects/Slicer-Release/DCMTK/dcmseg/include/dcmtk/dcmseg/segdoc.h:27,
                 from /home/jcfr/Projects/dcmqi/include/dcmqi/ImageSEGConverter.h:12,
                 from /home/jcfr/Projects/dcmqi/apps/seg/Testing/../segimage2itkimage.cxx:6,
                 from /home/jcfr/Projects/dcmqi/apps/seg/Testing/segimage2itkimageTest.cxx:10:
/home/jcfr/Projects/Slicer-Release/DCMTK/dcmiod/include/dcmtk/dcmiod/iodreferences.h:111:20: warning: ‘virtual bool IODReference::readFromFile(const OFString&)’ was hidden [-Woverloaded-virtual]
     virtual OFBool readFromFile(const OFString& filename);
                    ^
/home/jcfr/Projects/Slicer-Release/DCMTK/dcmiod/include/dcmtk/dcmiod/iodreferences.h:207:20: warning:   by ‘virtual bool IODImageReference::readFromFile(const OFString&, OFVector<unsigned int>)’ [-Woverloaded-virtual]
     virtual OFBool readFromFile(const OFString& filename,
                    ^
/home/jcfr/Projects/Slicer-Release/DCMTK/dcmiod/include/dcmtk/dcmiod/iodreferences.h:111:20: warning: ‘virtual bool IODReference::readFromFile(const OFString&)’ was hidden [-Woverloaded-virtual]
     virtual OFBool readFromFile(const OFString& filename);
                    ^
/home/jcfr/Projects/Slicer-Release/DCMTK/dcmiod/include/dcmtk/dcmiod/iodreferences.h:256:20: warning:   by ‘virtual bool IODSegmentationReference::readFromFile(const OFString&, OFVector<short unsigned int>)’ [-Woverloaded-virtual]
     virtual OFBool readFromFile(const OFString& filename,
                    ^

```

```
In file included from /home/jcfr/Projects/Slicer-Release/DCMTK/dcmsr/include/dcmtk/dcmsr/dsrdoctr.h:29:0,
                 from /home/jcfr/Projects/Slicer-Release/DCMTK/dcmsr/include/dcmtk/dcmsr/dsrdoc.h:29,
                 from /home/jcfr/Projects/dcmqi/apps/sr/tid1500reader.cxx:8:
/home/jcfr/Projects/Slicer-Release/DCMTK/dcmsr/include/dcmtk/dcmsr/dsrdocst.h:166:25: warning: ‘virtual OFCondition DSRDocumentSubTree::print(std::ostream&, size_t, const DSRPositionCounter*)’ was hidden [-Woverloaded-virtual]
     virtual OFCondition print(STD_NAMESPACE ostream &stream,
                         ^
In file included from /home/jcfr/Projects/Slicer-Release/DCMTK/dcmsr/include/dcmtk/dcmsr/dsrdoc.h:29:0,
                 from /home/jcfr/Projects/dcmqi/apps/sr/tid1500reader.cxx:8:
/home/jcfr/Projects/Slicer-Release/DCMTK/dcmsr/include/dcmtk/dcmsr/dsrdoctr.h:97:25: warning:   by ‘virtual OFCondition DSRDocumentTree::print(std::ostream&, size_t)’ [-Woverloaded-virtual]
     virtual OFCondition print(STD_NAMESPACE ostream &stream,
                         ^
```

**Comment by jcfr** [comment]
Look like circle ci is failing because of ... 
```
Fetched 8,833 kB in 4s (1,811 kB/s)
W: Failed to fetch https://apt.dockerproject.org/repo/dists/ubuntu-precise/main/binary-amd64/Packages  gnutls_handshake() failed: A TLS fatal alert has been received.

E: Some index files failed to download. They have been ignored, or old ones used instead.

sudo apt-get update returned exit code 100
```

**Comment by fedorov** [comment]
I restarted the build, but there is still the same error...

**Comment by jcfr** [comment]
Look like the solution it to switch to trusty images for Ubuntu: https://discuss.circleci.com/t/php-compiled-with-curl-gnutls-curl-fails-to-validate-some-certs/1641/4

**Comment by jcfr** [comment]
I will look into adding the version to the circle.yml

**Comment by jcfr** [comment]
I switched CircleCI to Ubuntu 14.04

**Comment by jcfr** [comment]
`apt-get update` now succeeds

**Comment by fedorov** [comment]
great! I wonder why it started happening just now

**Comment by jcfr** [comment]
> great! I wonder why it started happening just now

It is possible that `apt.dockerproject.org` updated their certificate and older implementation of TLS can not work with it ...

**Comment by fedorov** [comment]
makes sense!

**Comment by fedorov** [comment]
@jcfr `convert.py` is not used by any of the tests - any particular reason why you worry about it? It is a utility tool for helping converting XML files into the JSON contexts.

There are issues installing lxml on CircleCI, and I don't think it is important to distract on fixing those, if we don't need convert.py

**Comment by jcfr** [comment]
CircleCI is failing because lxml used to convert output of Ctest for Junit can not run.
See https://github.com/QIICR/dcmqi/blob/05f9fda7e9a0ba3b5c21bb42d250b5a121076869/circle.yml#L42

**Comment by fedorov** [comment]
Aaah, I see. I confused convert.py with another tool. I guess this is now the result of changing Lunux distro? 

Would it be easier to use xsltproc instead for applying this transformation? It might already be available on lunux http://www.sagehill.net/docbookxsl/InstallingAProcessor.html

**Comment by fedorov** [comment]
Indeed, I confirmed we can install xsltproc with apt-get.

File: circle.yml [hunk 1]
```diff
@@ -34,13 +37,13 @@ dependencies:
...
     # Use scikit-ci-addons to streamline installation of CMake
-    - sudo pip install scikit-ci-addons==0.11.0
+    - pip install scikit-ci-addons==0.11.0
     - ci_addons circle/install_cmake 3.6.2
 
   post:
     - mkdir -p ~/dcmqi-build && cd ~/dcmqi-build && cmake -G Ninja ../dcmqi && ninja
     # Install package required by convert.py
-    - sudo pip install lxml==3.6.4
+    - pip install lxml==3.6.4
```
Comment (jcfr) [inline circle.yml]: Moving forward, these could be added to a file named `requirements-circleci.txt`

Comment (jcfr) [inline circle.yml]: That would be installed in the pre section

Comment (jcfr) [inline circle.yml]: or inside a `requirements-dev.txt`

**Comment by fedorov** [comment]
Great, it works!

@michaelonken see above some warnings from dcmtk, in case you want to fix them. just fyi

**Comment by jcfr** [comment]
Merging. Travis was green prior to the last commit that is affecting on CircleCi

**Comment by michaelonken** [comment]
I fixed the warning in dcmiod and will notify Jörg about the it regarding dcmsr. I think the warning (hidden virtual function) can be ignored but at least for my code it's ok for now having the function non-virtual if it makes the compiler happy :)

**Comment by jcfr** [comment]
@michaelonken  Thanks for looking into this. If keeping the function virtual is important, I think the keyword `using` may be helpful.   Example: http://stackoverflow.com/questions/18515183/c-overloaded-virtual-function-warning-by-clang#18515300

**Comment by michaelonken** [comment]
Nice @jcfr , best solution for my code, I'll commit that one.

**Comment by michaelonken** [comment]
FYI: This has been fixed in DCMTK (commit [8604ee](http://git.dcmtk.org/?p=dcmtk.git;a=commit;h=8604ee6cd8b1deb3cf5d241c7957e03414c32e13) and [a0586c](http://git.dcmtk.org/?p=dcmtk.git;a=commit;h=a0586cb22239c6bf22b4221eebfeebcd1cfaa125).

**Pull request merged**